### PR TITLE
add that current block is in the same chain as finalized ancestor

### PR DESCRIPTION
**Pull request by djrtwo**
Address #1959 with the addition of a gossip validation for finalized ancestor.

Also move the proposer shuffling check to after the chain/ancestor checks. The proposer id can't be checked against the shuffling unless the chain that the block is in is known. So it is natural to have this condition after the ancestor checks

### CODE REVIEW

File: specs/phase0/p2p-interface.md [hunk 1]
```diff
@@ -295,15 +295,18 @@ The following validations MUST pass before forwarding the `signed_beacon_block`
   (a client MAY choose to validate and store such blocks for additional purposes -- e.g. slashing detection, archive nodes, etc).
 - _[IGNORE]_ The block is the first block with valid signature received for the proposer for the slot, `signed_beacon_block.message.slot`.
 - _[REJECT]_ The proposer signature, `signed_beacon_block.signature`, is valid with respect to the `proposer_index` pubkey.
+- _[IGNORE]_ The block's parent (defined by `block.parent_root`) has been seen
+  (via both gossip and non-gossip sources)
+  (a client MAY queue blocks for processing once the parent block is retrieved).
+- _[REJECT]_ The block's parent (defined by `block.parent_root`) passes validation.
+- _[REJECT]_ The current `finalized_checkpoint` is an ancestor of `block` -- i.e.
```
Comment (protolambda) [inline specs/phase0/p2p-interface.md:302]: We may need to mirror this to attestation validation? Or do we rely on forkchoice there?

Comment (djrtwo) [reply specs/phase0/p2p-interface.md:302]: added

File: specs/phase0/p2p-interface.md [hunk 2]
```diff
@@ -333,6 +336,10 @@ The following validations MUST pass before forwarding the `signed_aggregate_and_
   (via both gossip and non-gossip sources)
   (a client MAY queue aggregates for processing once block is retrieved).
 - _[REJECT]_ The block being voted for (`aggregate.data.beacon_block_root`) passes validation.
+- _[REJECT]_ The current `finalized_checkpoint` is an ancestor of the `block` defined by `aggregate.data.beacon_block_root` -- i.e.
+  `get_ancestor(store, hash_tree_root(block), compute_start_slot_at_epoch(store.finalized_checkpoint.epoch))
```
Comment (protolambda) [inline specs/phase0/p2p-interface.md:340]:
```suggestion
  `get_ancestor(store, aggregate.data.beacon_block_root, compute_start_slot_at_epoch(store.finalized_checkpoint.epoch))
```

File: specs/phase0/p2p-interface.md [hunk 3]
```diff
@@ -393,6 +400,11 @@ The following validations MUST pass before forwarding the `attestation` on the s
   (via both gossip and non-gossip sources)
   (a client MAY queue aggregates for processing once block is retrieved).
 - _[REJECT]_ The block being voted for (`attestation.data.beacon_block_root`) passes validation.
+- _[REJECT]_ The current `finalized_checkpoint` is an ancestor of the `block` defined by `attestation.data.beacon_block_root` -- i.e.
+  `get_ancestor(store, hash_tree_root(block), compute_start_slot_at_epoch(store.finalized_checkpoint.epoch))
```
Comment (protolambda) [inline specs/phase0/p2p-interface.md:404]:
```suggestion
  `get_ancestor(store, attestation.data.beacon_block_root, compute_start_slot_at_epoch(store.finalized_checkpoint.epoch))
```

**Pull request merged**